Objects/unionobject.c

```diff
@@ -242,7 +303,36 @@ _Py_union_type_or(PyObject* self, PyObject* other)
         Py_RETURN_NOTIMPLEMENTED;
     }
 
-    PyObject *tuple = PyTuple_Pack(2, self, other);
+    if (self == Py_None) {
```

Comment:
Does this happen? Isn't `self` guaranteed to be a `types.UnionType`?

Reply:
Actually, it's not always `type.UnionType`. For instance:
```py
None | int
```